pkg/network/go/bininspect/pclntab.go

```diff
@@ -0,0 +1,307 @@
+	switch {
+	case leMagic == go116magic:
+		p.byteOrderParser, p.cachedVersion = binary.LittleEndian, ver116
+	case beMagic == go116magic:
+		p.byteOrderParser, p.cachedVersion = binary.BigEndian, ver116
+	case leMagic == go118magic:
+		p.byteOrderParser, p.cachedVersion = binary.LittleEndian, ver118
+	case beMagic == go118magic:
+		p.byteOrderParser, p.cachedVersion = binary.BigEndian, ver118
+	case leMagic == go120magic:
+		p.byteOrderParser, p.cachedVersion = binary.LittleEndian, ver120
+	case beMagic == go120magic:
+		p.byteOrderParser, p.cachedVersion = binary.BigEndian, ver120
+	default:
+		return ErrUnsupportedPCLNTABVersion
+	}
+
+	p.ptrSize = uint32(pclntabHeader[7])
+	p.ptrBufferSizeHelper = make([]byte, p.ptrSize)
+
+	// offset is based on https://github.com/golang/go/blob/6a861010be9eed02d5285509cbaf3fb26d2c5041/src/debug/gosym/pclntab.go#L252
+	offset := func(word uint32) uint64 {
+		off := 8 + word*p.ptrSize
+		if n, err := p.section.ReadAt(p.ptrBufferSizeHelper, int64(off)); err != nil || n != int(p.ptrSize) {
+			return 0
+		}
+		return p.uintptr(p.ptrBufferSizeHelper)
+	}
+
+	switch p.cachedVersion {
+	case ver118, ver120:
+		p.funcTableSize = uint32(offset(0))
+		p.funcNameTable = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(3)),
+		}
+		p.funcData = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(7)),
+		}
+		p.funcTable = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(7)),
+		}
+	case ver116:
+		p.funcTableSize = uint32(offset(0))
+		p.funcNameTable = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(2)),
+		}
+		p.funcData = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(6)),
+		}
+		p.funcTable = sectionAccess{
+			section:    p.section,
+			baseOffset: int64(offset(6)),
+		}
+	}
+
+	return nil
+}
+
+// uintptr returns the pointer-sized value encoded at b.
+// The pointer size is dictated by the table being read.
+// based on https://github.com/golang/go/blob/6a861010be9eed02d5285509cbaf3fb26d2c5041/src/debug/gosym/pclntab.go#L186.
+func (p *pclntanSymbolParser) uintptr(b []byte) uint64 {
+	if p.ptrSize == 4 {
+		return uint64(p.byteOrderParser.Uint32(b))
+	}
+	return p.byteOrderParser.Uint64(b)
+}
+
+// getFuncTableFieldSize returns the size of a field in the function table.
+// based on https://github.com/golang/go/blob/6a861010be9eed02d5285509cbaf3fb26d2c5041/src/debug/gosym/pclntab.go#L388-L392
+func getFuncTableFieldSize(version version, ptrSize int) int {
+	if version >= ver118 {
+		return 4
+	}
+	return ptrSize
+}
+
+// getSymbols returns the symbols from the pclntab section that match the symbol filter.
+// based on https://github.com/golang/go/blob/6a861010be9eed02d5285509cbaf3fb26d2c5041/src/debug/gosym/pclntab.go#L300-L329
+func (p *pclntanSymbolParser) getSymbols() (map[string]*elf.Symbol, error) {
+	numWanted, _ := p.symbolFilter.getMinMaxLength()
```

Comment:
Should be p.symbolFilter.getNumWanted()?

Reply:
oof